wandb/apis/public/api.py

```diff
@@ -775,14 +791,19 @@ def project(self, name: str, entity: Optional[str] = None) -> "public.Project":
 
         Arguments:
             name: (str) The project name.
-            entity: (str) Name of the entity requested.  If None, will fall back to the
+            entity: (str) Name of the entity or organization requested.  If None, will fall back to the
                 default entity passed to `Api`.  If no default entity, will raise a `ValueError`.
 
         Returns:
             A `Project` object.
         """
+        if is_artifact_registry_project(name):
+            org = entity or ""
         if entity is None:
             entity = self.settings["entity"] or self.default_entity
+        # Fetch org entity for artifact registry projects
+        if is_artifact_registry_project(name):
+            entity = self._fetch_org_entity_for_registry(org, entity)
```

Comment:
what happens if `org` is `None` here? Per line 802, `entity` could be `None`.

Reply:
why do we need the org name to fetch the org entity, given the team name?

Reply:
`_fetch_org_entity_for_registry` accepts an org as input purely just to see if the org or org entity that the user inputted in is valid. So if no org is passed in, we just simply assuming the shorthand route and fetch the entity for the user and there is no check to also make sure that the user inputted the right org